spring-integration-amqp/src/main/java/org/springframework/integration/amqp/outbound/AmqpOutboundEndpoint.java

```diff
@@ -111,23 +104,17 @@ public void setExchangeName(String exchangeName) {
 		this.exchangeName = exchangeName;
 	}
 
-	/**
-	 * @deprecated in favor of {@link #setExpressionExchangeName}. Will be changed in a future release
-	 * to use an {@link Expression} parameter.
-	 * @param exchangeNameExpression the expression to set.
-	 */
-	@Deprecated
-	public void setExchangeNameExpression(String exchangeNameExpression) {
-		Assert.hasText(exchangeNameExpression);
-		this.exchangeNameExpression = expressionParser.parseExpression(exchangeNameExpression);
+	public void setExchangeNameExpression(Expression exchangeNameExpression) {
```

Comment:
M-m-m. That is exactly what I'm trying to do here:
* make `...Expression()` setter based on `Expression`
* deprecate `setExpression...()` setter.
* and after this our discussion introduce a new `...ExpressionString()` setter.

Am I missing anything else?